packages/openapi-fetch/test/index.test.ts

```diff
@@ -407,15 +407,48 @@ describe("client", () => {
         return async () => Promise.resolve(response);
       }
 
-      const baseData = { works: true };
-      const customBaseFetch = createCustomFetch(baseData);
-      const client = createClient<paths>({ fetch: customBaseFetch });
-      expect((await client.GET("/self")).data).toBe(baseData);
+      const customFetch = createCustomFetch({ works: true });
+      mockFetchOnce({ status: 200, body: "{}" });
+
+      const client = createClient<paths>({ fetch: customFetch });
+      const { data } = await client.GET("/self");
+
+      // assert data was returned from custom fetcher
+      expect(data).toEqual({ works: true });
+
+      // assert global fetch was never called
+      expect(fetchMocker).not.toHaveBeenCalled();
+    });
+
+    it("accepts a custom fetch function per-request", async () => {
+      function createCustomFetch(data: any) {
+        const response = {
+          clone: () => ({ ...response }),
+          headers: new Headers(),
+          json: async () => data,
+          status: 200,
+          ok: true,
+        } as Response;
+        return async () => Promise.resolve(response);
+      }
+
+      const fallbackFetch = createCustomFetch({ fetcher: "fallback" });
+      const overrideFetch = createCustomFetch({ fetcher: "override" });
+
+      mockFetchOnce({ status: 200, body: "{}" });
+
+      const client = createClient<paths>({ fetch: fallbackFetch });
+
+      // assert override function was called
+      const fetch1 = await client.GET("/self", { fetch: overrideFetch });
+      expect(fetch1.data).toEqual({ fetcher: "override" });
+
+      // assert fallback function still persisted (and wasn’t overridden)
+      const fetch2 = await client.GET("/self");
+      expect(fetch2.data).toEqual({ fetcher: "fallback" });
 
-      const data = { result: "it's working" };
-      const customFetch = createCustomFetch(data);
-      const customResponse = await client.GET("/self", { fetch: customFetch });
-      expect(customResponse.data).toBe(data);
+      // assert global fetch was never called
+      expect(fetchMocker).not.toHaveBeenCalled();
```

Comment:
This was an important part missing from the original test—testing that global `fetch` wasn’t actually invoked (we could assume, based on the data, but we weren’t explicitly checking).